tests/is_device_copyable/is_device_copyable_api.cpp

```diff
@@ -0,0 +1,161 @@
+  ~user_def_type() = default;
+
+  user_def_type(int _a, float _b, char _c) : a(_a), b(_b), c(_c) {}
+  user_def_type(const user_def_type& other)
+      : a(other.a), b(other.b), c(other.c) {}
+
+  user_def_type& operator=(const user_def_type& other) {
+    a = other.a;
+    b = other.b;
+    c = other.c;
+    return *this;
+  }
+
+  friend bool operator==(const user_def_type& lhs, const user_def_type& rhs) {
+    return (lhs.a == rhs.a && lhs.b == rhs.b && lhs.c == rhs.c);
+  }
+
+  int a;
+  float b;
+  char c;
+};
+
+template <>
+struct sycl::is_device_copyable<user_def_type> : std::true_type {};
+
+template <typename T>
+constexpr T expected_val() {
+  if constexpr (std::is_same_v<T, int>) {
+    return 42;
+  }
+  if constexpr (std::is_same_v<T, float>) {
+    return 21.f;
+  }
+  if constexpr (std::is_same_v<T, char>) {
+    return 'c';
+  }
+  return 0;
+}
+
+TEST_CASE("is_device_copyable UnaryTrait requirements",
+          "[is_device_copyable]") {
+  using device_copyable_t = sycl::is_device_copyable<user_def_type>;
+  using integral_constant_t = std::integral_constant<bool, true>;
+
+  // DefaultConstructible and CopyConstructible
+  CHECK(std::is_default_constructible_v<device_copyable_t>);
+  CHECK(std::is_copy_constructible_v<device_copyable_t>);
+
+  // Takes one template type parameter
+  CHECK(std::is_constructible_v<device_copyable_t>);
+
+  // Publicly and unambiguously derived from a specialization of
+  // integral_constant
+  constexpr bool is_base_of =
+      std::is_base_of_v<integral_constant_t, device_copyable_t>;
+  CHECK(is_base_of);
+
+  // The member names of the base characteristic are not hidden and are
+  // unambiguously available
+  CHECK(std::is_same_v<decltype(integral_constant_t::value),
+                       decltype(device_copyable_t::value)>);
+  CHECK(std::is_same_v<integral_constant_t::value_type,
+                       device_copyable_t::value_type>);
+  CHECK(std::is_same_v<decltype(integral_constant_t::value_type()),
+                       decltype(device_copyable_t::value_type())>);
+  CHECK(std::is_same_v<integral_constant_t::type, device_copyable_t::type>);
+  CHECK(integral_constant_t() == device_copyable_t());
+}
+
+TEST_CASE("is_device_copyable specialization for user defined class",
+          "[is_device_copyable]") {
+  INFO(
+      "User defined type must be device copyable after specialization for "
+      "sycl::is_device_copyable<T>");
+  REQUIRE(sycl::is_device_copyable<user_def_type>::value);
+
+  auto queue = sycl_cts::util::get_cts_object::queue();
+  const user_def_type reference_object(
+      expected_val<int>(), expected_val<float>(), expected_val<char>());
+
+  bool host_res;
+  user_def_type host_object(reference_object);
+  user_def_type device_object;
+  {
+    sycl::buffer<bool, 1> host_res_buf(&host_res, {1});
+    sycl::buffer<user_def_type, 1> host_data_buf(&host_object, {1});
+    sycl::buffer<user_def_type, 1> dev_data_buf(&device_object, {1});
```

Comment:
```suggestion
    sycl::buffer<bool> host_res_buf(&host_res, {1});
    sycl::buffer<user_def_type> host_data_buf(&host_object, {1});
    sycl::buffer<user_def_type> dev_data_buf(&device_object, {1});
```

Reply:
I don't think it was really necessary, but applied.